docs/services.md

```diff
@@ -51,4 +104,162 @@ A new service can be created by inheriting from `Node.Service` and implementing
 
 The `package.json` for the service module can either export the `Node.Service` directly, or specify a specific module to load by including `"bitcoreNode": "lib/bitcore-node.js"`.
 
-Please take a look at some of the existing services for implemenation specifics.
+Please take a look at some of the existing services for implementation specifics.
+
+### Adding an index
+
+One quite useful feature exposed to services is the ability to index arbitrary data in the blockchain. To do so we make
+use of leveldb, a simple key-value store. As a service we can expose a 'blockHandler' function which is called each time
+a new block is added or removed from the blockchain. This gives us access to every new transaction received, allowing 
+us to index them. Let's take a look at an example where we will index the time that a transaction was confirmed.
+
+```js
+//Index prefix, so that we can determine the difference between our index 
+//and the indexes provided by other services
+MyService.datePrefix = new Buffer('10', 'hex');
+
+MyService.minPosition = new Buffer('00000', 'hex');
+MyService.maxPosition = new Buffer('99999', 'hex');
+
+//This function is automatically called when a block is added or receieved
+MyService.prototype.prototype.blockHandler = function(block, addOutput, callback) {
+
+  //Determine if the block is added or removed, and therefore whether we are adding
+  //or deleting indexes
+  var databaseAction = 'put';
+  if (!addOutput) {
+    databaseAction = 'del';
+  }
+
+  //An array of all leveldb operations we will be committing
+  var operations = [];
+
+  //Timestamp of the current block
+  var blocktime = new Buffer(block.header.time);
```

Comment:
When reading the timestamp later, it could be useful here to use:
```
var buf = new Buffer(4)`; 
buf.writeUInt32BE(block.header.time);
```

It will take less space to store:
```
> var a = new Buffer('1445011783');
undefined
> a
<Buffer 31 34 34 35 30 31 31 37 38 33>
> var b = new Buffer(4);
undefined
> b.writeUInt32BE(1445011783);
4
> b
<Buffer 56 21 21 47>
```